src/amo/components/AddonRecommendations/index.js

```diff
@@ -0,0 +1,169 @@
+export const TAAR_COHORT_COOKIE_NAME = 'TAARCohort';
+export const TAAR_COHORT_INCLUDED: 'TAAR_COHORT_INCLUDED'
+  = 'TAAR_COHORT_INCLUDED';
+export const TAAR_COHORT_EXCLUDED: 'TAAR_COHORT_EXCLUDED'
+  = 'TAAR_COHORT_EXCLUDED';
+
+export type CohortName = typeof TAAR_COHORT_INCLUDED | typeof TAAR_COHORT_EXCLUDED;
+
+type Props = {|
+  cookie: typeof defaultCookie,
+  randomizer: () => number,
+  tracking: typeof defaultTracking,
+  addon: AddonType | null,
+  className?: string,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+  recommendations: Recommendations | null,
+|};
+
+
+export class AddonRecommendationsBase extends React.Component<Props> {
+  static defaultProps = {
+    cookie: defaultCookie,
+    randomizer: Math.random,
+    tracking: defaultTracking,
+    recommendations: null,
+  };
+
+  componentWillMount() {
+    const { cookie, randomizer, addon, recommendations } = this.props;
+
+    // Set a cohort for the experiment.
+    this.cohort = cookie.load(TAAR_COHORT_COOKIE_NAME);
+    if (this.cohort === undefined) {
+      this.cohort = randomizer() > 0.5 ?
+        TAAR_COHORT_INCLUDED : TAAR_COHORT_EXCLUDED;
+      cookie.save(TAAR_COHORT_COOKIE_NAME, this.cohort, { path: '/' });
+    }
+
+    if (!!addon && !recommendations) {
+      this.dispatchFetchRecommendations({
+        guid: addon.guid,
+        recommended: this.cohort === TAAR_COHORT_INCLUDED,
+      });
+    }
+  }
+
+  componentWillReceiveProps({
+    addon: newAddon,
+    recommendations: newRecommendations,
+  }: Props) {
+    const {
+      addon: oldAddon,
+      recommendations: oldRecommendations,
+      tracking,
+    } = this.props;
+
+    // Fetch recommendations when the add-on changes.
+    if (oldAddon !== newAddon && !!newAddon) {
+      this.dispatchFetchRecommendations({
+        guid: newAddon.guid,
+        recommended: this.cohort === TAAR_COHORT_INCLUDED,
+      });
+    }
+
+    // Send the GA ping when recommendations are loaded.
+    if (oldRecommendations !== newRecommendations && !!newRecommendations) {
+      const { fallbackReason, loading, outcome } = newRecommendations;
+
+      if (loading) {
+        return;
+      }
+
+      invariant(newAddon, 'newAddon is required');
+      invariant(outcome, 'outcome is required');
+
+      // TODO: Determine the exact format for this output.
+      const action = oneLine`outcome: ${outcome} |
+        fallbackReason: ${fallbackReason || ''}`;
+      tracking.sendEvent({
+        action,
+        category: TAAR_IMPRESSION_CATEGORY,
+        label: newAddon.name,
+      });
+    }
+  }
+
+  cohort: CohortName;
+
+  dispatchFetchRecommendations({ guid, recommended }: Object) {
+    this.props.dispatch(fetchRecommendations({
+      errorHandlerId: this.props.errorHandler.id,
+      guid,
+      recommended,
+    }));
+  }
+
+  render() {
+    const { className, i18n, recommendations } = this.props;
+
+    if (!recommendations) {
+      return null;
```

Comment:
Maybe we should add a `log.debug()` message here, like "No recommandations, hiding the `AddonRecommendations` component". What do you think?

Reply:
Sure.